lib/commands/performance.js

```
@@ -13,31 +15,179 @@ const PERF_RECORD_FEAT_NAME = 'perf_record';
+    this._timeout = (opts.timeout && opts.timeout > 0) ? opts.timeout : DEFAULT_TIMEOUT_MS;
+    this._profileName = opts.profileName || DEFAULT_PROFILE_NAME;
+    this._pid = opts.pid;
+    this._udid = udid;
+    this._logger = logger.getLogger(
+      `${_.truncate(this.profileName, {length: 10})}@${this.udid.substring(0, 8)}`);
+    this._archivePromise = null;
+  }
+
+  get profileName () {
+    return this._profileName;
+  }
+
+  async getOriginalReportPath () {
+    return (await fs.exists(this._reportPath)) ? this._reportPath : '';
+  }
+
+  async getZippedReportPath () {
+    if (await fs.exists(this._zippedReportPath)) {
+      return this._zippedReportPath;
+    }
+    const originalReportPath = await this.getOriginalReportPath();
+    if (!originalReportPath) {
+      return '';
+    }
+    const zippedReportPath = originalReportPath.replace(`.${DEFAULT_EXT}`, '.zip');
+    // This is to prevent possible race conditions, because the archive operation
+    // could be pretty time-intensive
+    if (!this._archivePromise) {
+      this._archivePromise = zip.toArchive(zippedReportPath, {
+        cwd: originalReportPath,
+      });
+    }
+    await this._archivePromise;
+    this._zippedReportPath = zippedReportPath;
+    return this._zippedReportPath;
+  }
+
+  isRunning () {
+    return !!(this._process?.isRunning);
+  }
+
+  async _enforceTermination () {
+    if (this._process && this.isRunning()) {
+      this._logger.debug('Force-stopping the currently running perf recording');
+      try {
+        await this._process.stop('SIGKILL');
+      } catch (ign) {}
+    }
+    this._process = null;
+    if (this._archivePromise) {
+      this._archivePromise
+        // eslint-disable-next-line promise/prefer-await-to-then
+        .then(async () => {
+          await fs.rimraf(this._zippedReportPath);
+          this._archivePromise = null;
+        })
+        .catch(() => {});
+    } else {
+      if (await fs.exists(this._zippedReportPath)) {
```

`fs.rimraf` silently passes over non-existent paths, so there is no need to check before calling.

---

does it handle an empty string as non existent path?

---

As long as it's a string. It can't be `null` or `undefined`.